thoth/common/openshift.py

```diff
@@ -1480,18 +1496,51 @@ def schedule_adviser(
                 "Unable to schedule adviser without backend namespace being set"
             )
 
-        job_id = job_id or self.generate_id("adviser")
-        parameters = locals()
-        parameters.pop("self", None)
-        return self._schedule_workload(
-            run_method_name=self.run_adviser.__name__,
-            run_method_parameters=parameters,
-            template_method_name=self.get_adviser_template.__name__,
-            job_id=job_id,
-            namespace=self.backend_namespace,
-            labels={"component": "adviser"},
+        if not self.use_argo:
+            job_id = job_id or self.generate_id("adviser")
+            parameters = locals()
+            parameters.pop("self", None)
+            return self._schedule_workload(
+                run_method_name=self.run_adviser.__name__,
+                run_method_parameters=parameters,
+                template_method_name=self.get_adviser_template.__name__,
+                job_id=job_id,
+                namespace=self.backend_namespace,
+                labels={"component": "adviser"},
+            )
+
+        import workflows
+        adviser_id = job_id or self.generate_id("adviser")
+        template_parameters = {}
+        template_parameters["THOTH_ADVISER_JOB_ID"] = adviser_id
+        template_parameters["THOTH_ADVISER_REQUIREMENTS"] = application_stack["requirements"]
+        template_parameters["THOTH_ADVISER_REQUIREMENTS_LOCKED"] = application_stack["requirements_lock"]
+        template_parameters["THOTH_ADVISER_LIBRARY_USAGE"] = json.dumps(library_usage)
+        template_parameters["THOTH_ADVISER_REQUIREMENTS_FORMAT"] = "pipenv"
+        template_parameters["THOTH_ADVISER_RECOMMENDATION_TYPE"] = recommendation_type
+        template_parameters["THOTH_ADVISER_RUNTIME_ENVIRONMENT"] = runtime_environment
+        template_parameters["THOTH_ADVISER_LIMIT"] = limit
+        template_parameters["THOTH_ADVISER_COUNT"] = count
+        template_parameters["THOTH_ADVISER_LIMIT_LATEST_VERSIONS"] = limit_latest_versions
+
+        workflow_parameters = {}
+        workflow_parameters["THOTH_DOCUMENT_ID"] = f"advise-{adviser_id}"
+
+        workflow_manager = workflows.WorkflowManager(
```

Comment:
I think this will be used more times, maybe worth to make it property or instance variable?